pkg/test/expectations/expectations.go

```diff
@@ -535,6 +540,34 @@ func ExpectNodeClaims(ctx context.Context, c client.Client) []*v1beta1.NodeClaim
 	return lo.ToSlicePtr(nodeClaims.Items)
 }
 
+func ExpectStateNodeExists(cluster *state.Cluster, node *v1.Node) *state.StateNode {
+	GinkgoHelper()
+	var ret *state.StateNode
+	cluster.ForEachNode(func(n *state.StateNode) bool {
```

Comment:
Wish we could just lo.Find(cluster.Nodes(), ...).

Reply:
+1

Reply:
Originally I had made a GetStateNode() that accesses the cluster.nodes map, but it was so long ago I forget what went wrong. I could add this to a list of followups to speed this up.